clojure-mode.el

```diff
@@ -1009,7 +1009,7 @@ nil."
                     (one-or-more (any whitespace "\n")))
       ;; why is this here? oh (in-ns 'foo) or (ns+ :user)
       (zero-or-one (any ":'"))
-      (group (one-or-more (not (any "()\"" whitespace))) word-end)))
+      (group (one-or-more (not (any "()\"" whitespace))) symbol-end)))
```

Comment:
The problem is that I tried using symbol instead of word semantics when I inherited the mode and everything broke, therefore my reservations... Using symbol semantics is the reasonable thing to do for pretty much everything, but I really things something was fucked up regarding them. I recall `lisp-mode`'s code was full of word references itself, which was odd. 

At any rate - a few unit tests won't hurt.